### PR TITLE
feat: add LEANN connector support

### PR DESCRIPTION
**Pull request by iamonuwa**
### Introduction
To contribute code to the Pathway project, start by discussing your proposed changes on Discord or by filing an issue.
Once approved, follow the fork + pull request model against the main branch, ensuring you've signed the contributor license agreement.

### Context
This PR adds a LEANN output connector (`pw.io.leann.write`) to support LEANN vector indices in RAG pipelines. LEANN is a storage-efficient vector database that uses graph-based selective recomputation to achieve 97% storage reduction compared to traditional solutions.

As discussed in issue #173, LEANN is implemented as a Python output connector following the established pattern from issue #167. The connector accumulates documents and builds the LEANN index on pipeline completion, with support for both HNSW and DiskANN backends.

### How has this been tested?
- **Unit tests** (`python/pathway/tests/test_io_leann.py`): Test observer behavior including additions, deletions, empty text handling, metadata extraction, and index build logic with mocked LEANN dependencies.
- **Integration tests** (`integration_tests/leann/test_leann.py`): End-to-end tests verifying index creation, search functionality, metadata columns, DiskANN backend, custom text columns, directory creation, and edge cases like empty tables.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:
- [x] My code follows the code style of this project,
- [x] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: python/pathway/io/leann/__init__.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+@trace_user_frame
+def write(
+    table: Table,
+    index_path: str | os.PathLike,
+    text_column: str = "text",
+    *,
+    metadata_columns: list[str] | None = None,
+    backend_name: Literal["hnsw", "diskann"] = "hnsw",
+    embedding_mode: str | None = None,
```
Comment (zxqfd555) [inline python/pathway/io/leann/__init__.py]: It can also be a literal.

**Review by zxqfd555 — COMMENTED** [review]
Thank you for your interest in Pathway!
I have left several comments: could you please address them?

**Comment by zxqfd555** [comment]
I think the change also deserves a mention in the CHANGELOG :)

**Comment by iamonuwa** [comment]
issues resolved https://github.com/pathwaycom/pathway/pull/183/commits/bc6044b470788a1baa4c07d4b1fc22eb67bf4114 👍🏾

File: python/pathway/io/leann/__init__.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    Returns:
+        None
+
+    Note:
+        - Rows with empty or null text are skipped with a warning logged.
+        - Existing index files are overwritten on each build.
+        - LEANN requires the ``leann`` package. Please visit
+          https://github.com/yichuan-w/LEANN for installation instructions.
+        - metadata_columns are collected but not yet indexed (pending LEANN API support).
```
Comment (zxqfd555) [inline python/pathway/io/leann/__init__.py]: At a glance, LEANN builder has a `metadata` parameter in `add_text`. Could you clarify more about the extent of the limitation? If it isn't supported yet, we can probably remove it so as not to mislead the users, and add a TODO.

Besides, if we have metadata columns, I would propose to use `ColumnReference` to pass them, same as I've proposed with `text_column`.

File: python/pathway/io/leann/__init__.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    Args:
+        table: The Pathway table to index.
+        index_path: Path where the LEANN index will be saved (.leann file).
+        text_column: Name of the column containing text to index.
+        metadata_columns: List of additional columns to include as metadata.
+        backend_name: LEANN backend - "hnsw" or "diskann".
+        embedding_mode: Embedding provider - "sentence-transformers", "openai",
+            "mlx", or "ollama".
+        embedding_model: Specific embedding model name (e.g., "facebook/contriever").
+        embedding_options: Additional embedding config (api_key, base_url, etc.).
```
Comment (zxqfd555) [inline python/pathway/io/leann/__init__.py]: Is there any list for what these parameters may be?
If there is no such list, we can just leave it as it is, otherwise I'd propose linking it.

**Comment by zxqfd555** [comment]
Closing the PR due to inactivty.
The feature will be integrated into the framework by Pathway team.
